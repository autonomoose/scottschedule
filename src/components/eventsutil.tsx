// events utilities and components
// exports default DisplayEvents,
//  also exports components CreateEvent, ModifyEvent
//  and data fetchEventsDB - full events
import React, { useEffect, useState } from 'react';
import { API } from 'aws-amplify';
import { useForm } from "react-hook-form";

import Box from '@mui/material/Box';
import Button from '@mui/material/Button';
import Card from '@mui/material/Card';
import IconButton from '@mui/material/IconButton';
import List from '@mui/material/List';
import ListItem from '@mui/material/ListItem';
import Typography from '@mui/material/Typography';

import { listEventsFull } from '../graphql/queries';
import { mutAddEvents, mutAddRules, mutDelEvents } from '../graphql/mutations';

// -------------------------------------------------
interface CreateEventProps {
  onComplete?: (status: string) => void,
  open: boolean
}
export const CreateEvent = (props: CreateEventProps) => {
    // form states
    const { register, handleSubmit, reset, formState } = useForm({
        defaultValues: {
            name: '',
            descr: '',
        }
    });
    const { isDirty, errors } = formState;

    interface FormNewEvParms {
        name: string,
        descr: string,
    };
    const formNewEvSubmit = async (data: FormNewEvParms) => {
        console.log('form data', data);
        try {
            const xdata = {'input': {
                'etype': 'ev',
                'evnames': data.name+"!args",
                'descr': data.descr,
                }
            };
            const result = await API.graphql({query: mutAddEvents, variables: xdata});
            console.log('updated', result);
            if (props.onComplete) {
                props.onComplete(data.name);
            }
        } catch (result) {
            console.log('failed update', result);
        }
    };

    return(
      <Box display={(props.open)?'block': 'none'}>
      <Card style={{marginTop: '3px', maxWidth: 350, minWidth: 350, flex: '1 1', background: '#FAFAFA',
       boxShadow: '-5px 5px 12px #888888', borderRadius: '0 0 5px 5px'}}>
        <Box mx={1}>
          <form key="newEv" onSubmit={handleSubmit(formNewEvSubmit)}>
          <Box display="flex" justifyContent="space-between" alignItems="baseline">
            <Typography variant='h6'>
              Add New Event
            </Typography>
          </Box>

          <Box><label>
            Name <input type="text" size={12} data-testid="nameInput"
             {...register('name', { required: true, pattern: /\S+/, maxLength:16 })}
             aria-invalid={errors.name ? "true" : "false"}
            />
          </label></Box>
          <Box><label> Description
            <input type="text" size={30} data-testid="descrInput"
             {...register('descr', { required: true, pattern: /\S+/, maxLength:30 })}
             aria-invalid={errors.descr ? "true" : "false"}
            />
          </label></Box>

          <Box mt={2} display='flex' justifyContent='flex-end'>
            <Button size="small" variant="outlined" onClick={() => reset()} disabled={!isDirty}>Reset</Button>
            <Button size="small" variant="contained" color="primary" type="submit" disabled={!isDirty}>Save</Button>
          </Box>

          </form>
        </Box>
      </Card></Box>
) };

// -------------------------------------------------
interface CreateRuleProps {
  evName: string,
  onComplete?: (status: string) => void,
  open: boolean
}
export const CreateRule = (props: CreateRuleProps) => {
    // form states
    const { register, handleSubmit, reset, formState, watch } = useForm({
        defaultValues: {
            cmd: 'begin',
            options: '',
            content: '',
        }
    });
    const { isDirty, errors } = formState;

    const formNewRuleCancel = async () => {
        if (props.onComplete) {
            props.onComplete('');
        }
    }
    interface FormNewRuleParms {
        cmd: string,
        options: string,
        content: string,
    };
    const formNewRuleSubmit = async (data: FormNewRuleParms) => {
        console.log('form data', data);
        try {
            let wkEvNames = props.evName+"!"+data.cmd;
            if (data.options !== '') {
                wkEvNames += ' ' + data.options;
            }
            const xdata = {'input': {
                'etype': 'ev',
                'evnames': wkEvNames,
                'rules': data.content,
                }
            };
            const result = await API.graphql({query: mutAddRules, variables: xdata});
            console.log('updated rules', result);
            if (props.onComplete) {
                props.onComplete(props.evName);
            }

        } catch (result) {
            console.log('failed update rules', result);
            if (props.onComplete) {
                props.onComplete('');
            }
        }
    };

    return(
      <Box display={(props.open)?'block': 'none'}>
      <Card style={{marginTop: '3px', maxWidth: 350, minWidth: 350, flex: '1 1', background: '#FAFAFA',
       boxShadow: '-5px 5px 12px #888888', borderRadius: '0 0 5px 5px'}}>
        <Box mx={1}>
          <form key="newRule" onSubmit={handleSubmit(formNewRuleSubmit)}>
          <Box display="flex" justifyContent="space-between" alignItems="baseline">
            <Typography variant='h6'>
              Add New Rule
            </Typography>
          </Box>

          <Box display='flex'>
            <select {...register("cmd")}>
              <option value="begin">begin</option>
              <option value="option">option</option>
            </select>
            <input disabled={(watch('cmd') === 'begin')} data-testid="optionsInput" {...register('options')} />
          </Box>

          <Box>
            <textarea rows={2} cols={27} data-testid="contentRule"
             {...register('content', { required: true,})}
             aria-invalid={errors.content ? "true" : "false"}
            ></textarea>
          </Box>

          <Box mt={2} display='flex' justifyContent='flex-end'>
            <Button size="small" variant="outlined" color='error' onClick={() => formNewRuleCancel()}>Cancel</Button>
            <Button size="small" variant="outlined" onClick={() => reset()} disabled={!isDirty}>Reset</Button>
            <Button size="small" variant="contained" type="submit" disabled={!isDirty}>Save</Button>
          </Box>

          </form>
        </Box>
      </Card></Box>
) };

// -------------------------------------------------
interface ModifyEventProps {
  evid: string,
  tasks: iTask,
  onComplete?: (status: string) => void,
  open: boolean,
}
export const ModifyEvent = (props: ModifyEventProps) => {
    const allTasks = props.tasks;
    const evid = props.evid;

    const [evRule, setEvRule] = useState('');

    // form states
    const { register, handleSubmit, reset, formState } = useForm({
        defaultValues: {
            descr: '',
        }
    });
    const { isDirty, errors } = formState;

    useEffect(() => {
        const defaultValues = {
            descr: (allTasks[evid] && allTasks[evid].descr)? allTasks[evid].descr : '',
        }
        reset(defaultValues);
    }, [allTasks, evid] );

    interface FormModEvParms {
        descr: string,
    };
    const formModEvSubmit = async (data: FormModEvParms) => {
        console.log('modform data', data);
        try {
            const xdata = {'input': {
                'etype': 'ev',
                'evnames': evid+"!args",
                'descr': data.descr,
                }
            };
            const result = await API.graphql({query: mutAddEvents, variables: xdata});
            console.log('updated', result);
            if (props.onComplete) {
                props.onComplete(evid);
            }
        } catch (result) {
            console.log('failed update', result);
        }
    };
    interface FormDelEventParms {
        cmd: string,
    };
    const formDelEvent = async (data: FormDelEventParms) => {
        console.log('formDel parms', data);
        try {
            const xdata = {'input': {
                'etype': 'ev',
                'evnames': evid+"!"+data.cmd,
                }
            };
            console.log('deleting', xdata);
            const result = await API.graphql({query: mutDelEvents, variables: xdata});
            console.log('deleted', result);
            if (props.onComplete) {
                props.onComplete(evid);
            }
        } catch (result) {
            console.log('failed delete', result);
        }
    };
    const formCallback = (status: string) => {
        console.log("mod callback status", status);
        setEvRule('');
        if (props.onComplete && status !== '') {
            props.onComplete(status);
        }
    };

    return(
      <Box display={(props.open)?'block': 'none'}>
      <Card style={{marginTop: '3px', maxWidth: 350, minWidth: 350, flex: '1 1', background: '#FAFAFA',
       boxShadow: '-5px 5px 12px #888888', borderRadius: '0 0 5px 5px'}}>
        <Box mx={1}>
          <form key="newEv" onSubmit={handleSubmit(formModEvSubmit)}>
          <Box display="flex" justifyContent="space-between" alignItems="baseline">
            <Typography variant='h6'>Modify Event</Typography>
          </Box>
          <Box display='flex' alignItems='center'>
            {evid}
            { (allTasks[evid] && allTasks[evid].schedRules.length === 0) &&
            <IconButton size='small' color='error' onClick={() => formDelEvent({'cmd': 'args'})}>X</IconButton>
            }
          </Box>

          <Box><label>
            <input type="text" size={30} data-testid="descrInput"
             {...register('descr', { required: true, pattern: /\S+/, maxLength:30 })}
             aria-invalid={errors.descr ? "true" : "false"}
            />
          </label></Box>

          <Box mt={2} display='flex' justifyContent='flex-end'>
            <Button size="small" variant="outlined" onClick={() => reset()} disabled={!isDirty}>Reset</Button>
            <Button size="small" variant="contained" color="primary" type="submit" disabled={!isDirty}>Save</Button>
          </Box>
          </form>
          <Box mb={1} display='flex' justifyContent='space-around'>
            <span>Rules ({(allTasks && allTasks[evid])? allTasks[evid].schedRules.length: 0})</span>
            <Button onClick={() => setEvRule(evid)}  size="small" variant="outlined" color="primary">New Rule</Button>
          </Box>
          {(allTasks && allTasks[evid]) &&
          <Box>
            <CreateRule evName={evRule} onComplete={formCallback} open={(evRule !== '')} />
            {
              allTasks[evid].schedRules.map((task: string) => {
                // get the cmd words out
                let words = task.split(' ');
                let cmd = words.shift() || 'begin';
                if (cmd === 'option') {
                    cmd = 'option ' +  words.shift() || '';
                }
                const displayRule = words.join(' ').split(',').join(', ');

              return(
                <Box key={`${evid}${cmd}`}  display='flex'>
                  <Box display='flex'>
                    <IconButton size='small' color='error' onClick={() => formDelEvent({cmd})}>X</IconButton>
                    {cmd}
                  </Box>
                  <Box mx={1} mb={1} px={1} sx={{ border: '1px solid grey' }} key={`${evid}${cmd}`}>
                    {displayRule}
                  </Box>
                </Box>
              )})
            }

          </Box>
          }

        </Box>
      </Card></Box>
) };

// -------------------------------------------------
interface DisplayEventProps {
  tasks: iTask,
  select?: (evid: string) => void,
}
const DisplayEvents = (props: DisplayEventProps) => {
    const allTasks = props.tasks;

    return(
      <List disablePadding dense sx={{marginLeft: '1em'}}>
        {
          Object.keys(allTasks).map((evid: string) => {
          return(
            <ListItem button key={evid} onClick={() => {props.select?.(evid);}}>
              {evid} - {allTasks[evid].descr}
            </ListItem>
           )})
        }
      </List>
) };

// -------------------------------------------------
export const fetchEventsDB = async (): Promise<iTask> => {
    try {
        const result: any = await API.graphql({query: listEventsFull})
        console.log("events loaded:", result.data.listEvents.items.length);

        const compactTasks = result.data.listEvents.items.reduce((resdict: iTask, item: iTaskDb) => {
            const evkeys = item.evnames.split('!');
            if (!resdict[evkeys[0]]) {
                resdict[evkeys[0]] = {descr: '', schedRules: []};
            }
            if (evkeys[1] === 'args') {
                resdict[evkeys[0]].descr = (item.descr)? item.descr: '';
            } else {
                resdict[evkeys[0]].schedRules.push(evkeys[1] + " " + item.rules);
            }
            return resdict;
        }, {});

        return(compactTasks);
    } catch (result) {
        console.log("got error", result);
        return({});
    }
};

export default DisplayEvents
