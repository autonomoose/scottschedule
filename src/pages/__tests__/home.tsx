import React from "react";
import { render, screen, waitFor } from "@testing-library/react";

import HomePage from "../home";
import { fetchEventsDB } from '../../components/eventsutil';
import { fetchSchedGroupsDB } from '../../components/schedgrputil';

jest.mock('../../components/eventsutil', () => ({
    fetchEventsDB: jest.fn(),
}));
jest.mock('../../components/schedgrputil', () => ({
    fetchSchedGroupsDB: jest.fn(),
}));

const mockEnqueue = jest.fn();
jest.mock('notistack', () => ({
  ...jest.requireActual('notistack'),
  useSnackbar: () => {
    return {
      enqueueSnackbar: mockEnqueue
    };
  }
}));

Date.now = jest.fn(() => 1482363367071);

describe("HomePage", () => {
  it("renders snapshot correctly", async () => {
    const {container, getByTestId} = render(<HomePage />);
    await waitFor(() => {
        expect(getByTestId('dataBackdrop')).not.toBeVisible();
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
