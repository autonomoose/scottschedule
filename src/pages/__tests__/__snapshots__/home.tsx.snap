// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage renders snapshot correctly 1`] = `
<div
  data-testid="authenticator"
>
  <div
    style="text-align: center;"
  >
    <div
      style="margin: 1rem auto; min-height: 100vh; background-color: rgb(238, 238, 238); text-align: left;"
    >
      <div>
        <header
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-1hc7nu0-MuiPaper-root-MuiAppBar-root"
          style="height: 54px; box-shadow: 0 1px 20px #000000; border-radius: 0 0 5px 5px;"
        >
          <div
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1kf0d50-MuiToolbar-root"
          >
            <button
              aria-label="Open menu"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeStart MuiIconButton-sizeMedium css-9sors8-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                height="40"
                viewBox="0 0 100 80"
                width="40"
              >
                <rect
                  height="10"
                  width="80"
                  x="10"
                  y="10"
                />
                <rect
                  height="10"
                  width="80"
                  x="10"
                  y="30"
                />
                <rect
                  height="10"
                  width="80"
                  x="10"
                  y="50"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <a
              href="/home"
            >
              <div
                class="MuiBox-root css-k008qs"
              >
                <img
                  alt="Werner Digital"
                  src="../images/wernerdigital-hosted.png"
                  width="26"
                />
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  Scottschedule
                </p>
              </div>
            </a>
            <div
              class="MuiBox-root css-1r3535z"
            >
              <a
                href="/usermaint"
                style="margin-right: 1em;"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  mockuname
                </p>
              </a>
              <button
                class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Sign Out
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </header>
        <div
          class="MuiDrawer-root MuiDrawer-docked css-7ik032-MuiDrawer-docked"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-12i7wg6-MuiPaper-root-MuiDrawer-paper"
            style="visibility: hidden; webkit-transform: translateX(-0px); transform: translateX(-0px);"
          >
            <div
              style="width: 178px; height: 39px; padding: 8px 8px; box-shadow: 0 1px 20px #000000; border-radius: 0 0 5px 5px;"
            >
              <div
                class="MuiBox-root css-xwg089"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
                >
                  MainÂ Menu
                </h6>
                <button
                  aria-label="Close menu"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    height="40"
                    viewBox="0 0 100 80"
                    width="40"
                  >
                    <polygon
                      points="70 10,10 35, 70 65"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
            />
            <ul
              class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
            >
              <a
                href="/home"
              >
                <div
                  class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-bshv44-MuiButtonBase-root-MuiListItem-root"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="HomeIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                      />
                    </svg>
                  </div>
                  <div
                    class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                    >
                      Home
                    </span>
                  </div>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </div>
              </a>
              <a
                href="/usermaint"
              >
                <div
                  class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-bshv44-MuiButtonBase-root-MuiListItem-root"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="StarIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                      />
                    </svg>
                  </div>
                  <div
                    class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                    >
                      Account
                    </span>
                  </div>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </div>
              </a>
              <a
                href="/scheds"
              >
                <div
                  class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-bshv44-MuiButtonBase-root-MuiListItem-root"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="StarIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                      />
                    </svg>
                  </div>
                  <div
                    class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                    >
                      Schedules
                    </span>
                  </div>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </div>
              </a>
              <a
                href="/events"
              >
                <div
                  class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-bshv44-MuiButtonBase-root-MuiListItem-root"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="StarIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                      />
                    </svg>
                  </div>
                  <div
                    class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                    >
                      Events
                    </span>
                  </div>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </div>
              </a>
              <a
                href="/help"
              >
                <div
                  class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-bshv44-MuiButtonBase-root-MuiListItem-root"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="StarIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                      />
                    </svg>
                  </div>
                  <div
                    class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                    >
                      Help
                    </span>
                  </div>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </div>
              </a>
              <div
                class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-bshv44-MuiButtonBase-root-MuiListItem-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="StarIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    Logout
                  </span>
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </ul>
          </div>
        </div>
      </div>
      <div
        style="margin: 0px auto; padding: 50px 0.5rem 1.45rem; max-width: 960px; color: rgb(0, 0, 0);"
      >
        <main>
          <div
            data-testid="mainPageDisplay"
          >
             
            <div
              class="MuiBox-root css-7kebnq"
            >
              <div
                style="float: left;"
              >
                <nav
                  aria-label="Breadcrumb"
                  class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
                >
                  <ol
                    class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
                  >
                    <li
                      class="MuiBreadcrumbs-li"
                    >
                      <span>
                        Home
                      </span>
                    </li>
                  </ol>
                </nav>
              </div>
              <div>
                <a
                  href="/help/home"
                >
                  Help
                </a>
              </div>
            </div>
            <div
              class="MuiBox-root css-1b1zfbw"
            >
              <div
                class="MuiBox-root css-0"
              >
                <div
                  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
                  style="max-width: 432px; min-width: 394px; flex: 1 1; background: rgb(245, 245, 230); box-shadow: 5px 5px 12px #888888; border-radius: 0 0 5px 5px;"
                >
                  <div
                    class="MuiBox-root css-dduipi"
                  />
                  <div
                    class="MuiBox-root css-1hyfx7x"
                  />
                  <div
                    class="MuiBox-root css-1iuufgg"
                  >
                    <div
                      class="MuiBox-root css-x1sij0"
                    >
                      <button
                        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                        tabindex="0"
                        type="button"
                      >
                        <h4
                          class="MuiTypography-root MuiTypography-h4 css-ki721n-MuiTypography-root"
                          id="mainclock"
                        >
                          00:00
                        </h4>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                      <h6
                        class="MuiTypography-root MuiTypography-subtitle1 css-1ystea6-MuiTypography-root"
                        id="mainpm"
                      >
                        PM
                      </h6>
                    </div>
                    <div
                      class="MuiBox-root css-pf0lra"
                    >
                      <h6
                        class="MuiTypography-root MuiTypography-h6 css-16ht8xe-MuiTypography-root"
                        id="maindate"
                      >
                        01/01/00
                      </h6>
                      <span
                        class="MuiTypography-root MuiTypography-caption css-1l11w9n-MuiTypography-root"
                        id="grouptitle"
                      >
                        group title
                      </span>
                    </div>
                    <div
                      class="MuiBox-root css-0"
                      id="status"
                    >
                      <div
                        class="MuiFormControl-root MuiFormControl-marginDense MuiTextField-root css-khqvv2-MuiFormControl-root-MuiTextField-root"
                      >
                        <label
                          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
                          data-shrink="false"
                          for="schedgroup"
                          id="schedgroup-label"
                        >
                          Schedule Group
                        </label>
                        <div
                          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                        >
                          <div
                            aria-expanded="false"
                            aria-haspopup="listbox"
                            aria-labelledby="schedgroup-label schedgroup"
                            class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-jedpe8-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                            id="schedgroup"
                            role="button"
                            tabindex="0"
                          >
                            <span
                              class="notranslate"
                            >
                              â
                            </span>
                          </div>
                          <input
                            aria-hidden="true"
                            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                            data-testid="schedgroup"
                            tabindex="-1"
                            value=""
                          />
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                            data-testid="ArrowDropDownIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M7 10l5 5 5-5z"
                            />
                          </svg>
                          <fieldset
                            aria-hidden="true"
                            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                          >
                            <legend
                              class="css-173wfxe"
                            >
                              <span>
                                Schedule Group
                              </span>
                            </legend>
                          </fieldset>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiBox-root css-5uisrr"
                  >
                    <button
                      class="MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root  css-1i749hr-MuiButtonBase-root-MuiButton-root"
                      tabindex="0"
                      type="button"
                    >
                      Off
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                    <button
                      class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButtonBase-root  css-9y1egq-MuiButtonBase-root-MuiButton-root"
                      tabindex="0"
                      type="button"
                    >
                      tomorrow
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <div
                    class="MuiBox-root css-1iie599"
                  />
                  <hr
                    class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
                  />
                  <div
                    class="MuiBox-root css-8ljrm5"
                  >
                    Test 
                    <audio
                      class="audio-element"
                      controls=""
                    >
                      <source
                        src="test-file-stub"
                        type="audio/wav"
                      />
                      Your browser doesn't support audio
                    </audio>
                  </div>
                  <div
                    class="MuiBox-root css-0"
                  >
                    <h4
                      class="MuiTypography-root MuiTypography-h4 css-5lfyn9-MuiTypography-root"
                    >
                      No Events found
                    </h4>
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                    >
                      To get started, go to the
                    </p>
                    <a
                      href="/events"
                    >
                      Events page
                    </a>
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                    >
                      and build your first Event!
                    </p>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="MuiBackdrop-root css-1178cxv-MuiBackdrop-root"
              style="opacity: 0; visibility: hidden;"
            >
              <span
                class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorInherit css-62e83j-MuiCircularProgress-root"
                data-testid="dataBackdrop"
                role="progressbar"
                style="width: 40px; height: 40px;"
              >
                <svg
                  class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                  viewBox="22 22 44 44"
                >
                  <circle
                    class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                    cx="44"
                    cy="44"
                    fill="none"
                    r="20.2"
                    stroke-width="3.6"
                  />
                </svg>
              </span>
            </div>
             
          </div>
           
        </main>
        <footer
          style="padding-top: 40px;"
        >
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
          />
          <div
            class="MuiBox-root css-1bnxa12"
          >
            <a
              href="/home"
            >
              Home
            </a>
            <a
              href="/usermaint"
            >
              Account
            </a>
            <a
              href="/scheds"
            >
              Schedules
            </a>
            <a
              href="/events"
            >
              Events
            </a>
            <a
              href="/help"
            >
              Help
            </a>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
          />
          <span
            class="MuiTypography-root MuiTypography-caption css-3merus-MuiTypography-root"
          >
            Â© 2021-
            2022
            , Werner Digital Technology Inc
          </span>
        </footer>
      </div>
    </div>
  </div>
</div>
`;
